tutorials/publish_subscribe/c/home_automation_publisher.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+ *
+ *  RTI grants Licensee a license to use, modify, compile, and create derivative
+ *  works of the Software.  Licensee has the right to distribute object form
+ *  only for use with RTI products.  The Software is provided "as is", with no
+ *  warranty of any type, including any warranty for fitness for any purpose.
+ *  RTI is under no obligation to maintain or support the Software.  RTI shall
+ *  not be liable for any incidental or consequential damages arising out of the
+ *  use or inability to use the software.
+ */
+
+#include "ndds/ndds_c.h"
+#include "home_automation.h"
+#include "home_automationSupport.h"
+
+static int publisher_shutdown(
```

Comment:
let's forward-declare `publisher_shutdown` and define it later. I would like `publish_sensor` to be the first thing